File: DependencyInjection/DependencyInjectionAfter/_1_ControlFreakRefactoredToDependencyInjection/Services/MsSqlBasedRepository.cs
using DependencyInjectionAfter._1_ControlFreakRefactoredToDependencyInjection.Interfaces;

namespace DependencyInjectionAfter._1_ControlFreakRefactoredToDependencyInjection.Services
{
  internal interface IRepository
  {
    void Save(AcmeMessage message);
  }

  class MsSqlBasedRepository : IRepository
  {
    private readonly DataDestination _sqlDataDestination;

    public MsSqlBasedRepository(DataDestination sqlDataDestination)
    {
      _sqlDataDestination = sqlDataDestination;
    }

    public void Save(AcmeMessage message)
    {
      message.WriteTo(_sqlDataDestination);
    }
  }
}

File: Java/dependency-injection/control-freak-refactored/src/main/java/com/github/grzesiek_galezowski/DependencyInjectionAfter/Services/MsSqlBasedRepository.java
package com.github.grzesiek_galezowski.DependencyInjectionAfter.Services;

import com.github.grzesiek_galezowski.DependencyInjectionAfter.Interfaces.AcmeMessage;
import com.github.grzesiek_galezowski.DependencyInjectionAfter.Interfaces.DataDestination;

public class MsSqlBasedRepository implements IRepository {
  private final DataDestination _sqlDataDestination;

  public MsSqlBasedRepository(DataDestination sqlDataDestination) {
    _sqlDataDestination = sqlDataDestination;
  }

  public void Save(AcmeMessage message) {
    message.WriteTo(_sqlDataDestination);
  }
}
